File: Liftr/CompactCountFormat.swift
import Foundation

func formatCompactCount(_ value: Int) -> String {
    formatCompactCount(Int64(value))
}

func formatCompactCount(_ value: Int64) -> String {
    if value >= 1_000_000 { return String(format: "%.1fM", Double(value) / 1_000_000) }
    if value >= 1_000 { return String(format: "%.1fk", Double(value) / 1_000) }
    return "\(value)"
}

File: android/app/src/main/java/com/lilru/liftr/domain/CompactCountFormat.kt
package com.lilru.liftr.domain

import java.util.Locale

fun formatCompactCount(value: Int): String = formatCompactCount(value.toLong())

fun formatCompactCount(value: Long): String = when {
    value >= 1_000_000L -> String.format(Locale.US, "%.1fM", value / 1_000_000.0)
    value >= 1_000L -> String.format(Locale.US, "%.1fk", value / 1_000.0)
    else -> value.toString()
}
